### PR TITLE
libbitstream: enable fallthrough switch case

### PR DESCRIPTION
**Pull request by tswhison**
Add cmake C compiler check (C++ TBD) for -Werror=implicit-fallthrough=3.
With this option, placing a /* FALLTHROUGH */ comment in the switch
case in question avoids what would otherwise be a warning.

### CODE REVIEW

File: libbitstream/bitstream.c [hunk 1]
```diff
@@ -177,17 +177,8 @@ STATIC void *opae_bitstream_parse_metadata(const char *metadata,
 	// version to 640/650 respectively.
 	// Allow 640/650 to serve as an alias for 1.
 	case 650:
```
Comment (asgibbons) [inline libbitstream/bitstream.c]: Don't need a fallthrough here?

Comment (tswhison) [reply libbitstream/bitstream.c]: It didn't seem to require it.  This was with gcc 8.x on Fedora 29.

File: cmake/modules/compiler_config.cmake [hunk 2]
```diff
@@ -63,6 +64,13 @@ set(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-g -O2 -Wall -Wextra -Werror")
 set(CMAKE_C_FLAGS_MINSIZEREL       "-Os -Wall -Wextra -Werror")
 set(CMAKE_CXX_FLAGS_MINSIZEREL     "-Os -Wall -Wextra -Werror")
 
+if(COMPILER_SUPPORTS_IMPLICIT_FALLTHROUGH3)
```
Comment (chipb) [inline cmake/modules/compiler_config.cmake]: The GCC documentation indicates `-Wimplicit-fallthrough=3` is enabled when `-Wextra` is enabled (as it already seems to be on e.g. line 55 and 64). Can you drop this extra check?

Comment (tswhison) [reply cmake/modules/compiler_config.cmake]: I tried it without, but it didn't seem to work.  The error message indicated that -Werror=implicit-fallthrough= (without the 3) was active.

Comment (chipb) [reply cmake/modules/compiler_config.cmake]: Whoops. Sorry I missed your PR opening comment, but thanks for double-checking!

**Pull request merged**